community/shell/src/main/java/org/neo4j/shell/StartClient.java

```diff
@@ -504,4 +500,29 @@ private static String pad( String string, int length )
         }
         return string;
     }
+
+    private static class ShellExecutionFailureException extends RuntimeException
+    {
+        private final Throwable cause;
+        private final Args args;
+
+        ShellExecutionFailureException( Throwable cause, Args args )
+        {
+            this.cause = cause;
+            this.args = args;
+        }
+
+        void dumpMessage( PrintStream out, PrintStream err )
+        {
+            String message = cause.getCause() instanceof ConnectException ?
+                             "Connection refused" : cause.getMessage();
+            err.println( "ERROR (-v for expanded information):\n\t" + message );
+            if ( args.has( "v" ) )
```

Comment:
The only time we use this function is in our own `main` method, where we do want to print. So I'll indicate this by making it `private` instead.